### PR TITLE
Add a SwiftPM build plugin to generate Swift wrappers for Java classes

### PR DESCRIPTION
**Pull request by DougGregor**
Introduce the plugin `Java2SwiftPlugin`, which will invoke Java2Swift to generate Swift wrappers for the specified Java classes. Each SwiftPM target that uses this plugin must contain a file named `Java2Swift.config` that specifies options for the Java2Swift tool. This is a JSON file with a top-level dictionary that currently can contain keys for the Java class path (`classPath`) and a dictionary called `classes` whose keys are Java canonical class names (such as `java.util.Vector`) and whose values are the corresponding Swift names. For example:

```swift
{
  "classes" : {
    "java.util.Vector" : "JavaVector"
  }
}

```

will translate the class `java.util.Vector` into the Swift type `JavaVector`. The resulting Swift files will become part of the Swift module that uses this plugin.

I've updated the JavaKitSampleApp to make use of this, albeit in a very narrow way that is not at all compelling. The user guide also needs updating.

Fixes issue #16

### CODE REVIEW

File: Samples/JavaKitSampleApp/Sources/JavaKitExample/Java2Swift.config [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "classes" : {
+    "java.util.Vector" : null
```
Comment (ktoso) [inline Samples/JavaKitSampleApp/Sources/JavaKitExample/Java2Swift.config]: Vector is a rarely used type that folks generally avoid as it predates the "proper collections framework", I'd recommend avoiding it in examples (it'll look weird to Java developers that we're using that anywhere) and using `java.util.ArrayList` in examples.

--- 

double checked in docs:

> As of the Java 2 platform v1.2, this class was retrofitted to implement the [List](https://docs.oracle.com/javase/8/docs/api/java/util/List.html) interface, making it a member of the [Java Collections Framework](https://docs.oracle.com/javase/8/docs/technotes/guides/collections/index.html). Unlike the new collection implementations, Vector is synchronized. If a thread-safe implementation is not needed, it is recommended to use [ArrayList](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html) in place of Vector.

File: Package.swift [hunk 2]
```diff
@@ -56,6 +56,11 @@ let package = Package(
       targets: ["JavaKit"]
     ),
 
+    .library(
+      name: "JavaRuntime",
+      targets: [ "JavaRuntime"]
```
Comment (ktoso) [inline Package.swift]:
```suggestion
      targets: ["JavaRuntime"]
```

Need to enable the Swift formatting enforcement soon 🤔

**Review by ktoso — APPROVED** [review]
Good start, LGTM, minor nitpick inline

File: .licenseignore [hunk 3]
```diff
@@ -5,6 +5,7 @@
 .swift-format
 .github/*
 *.md
+**/*.config
```
Comment (ktoso) [inline .licenseignore:8]: This is to fix the license checking CI job, since it checked and failed on the new file not having a license header

**Review by ktoso — APPROVED** [review]

File: Package.swift (hunk 2 above)
Comment (DougGregor) [reply Package.swift]: Eh, sure

File: .licenseignore (hunk 3 above)
Comment (DougGregor) [reply .licenseignore:8]: Ah, thank you!

File: Samples/JavaKitSampleApp/Sources/JavaKitExample/Java2Swift.config (hunk 1 above)
Comment (DougGregor) [reply Samples/JavaKitSampleApp/Sources/JavaKitExample/Java2Swift.config]: I was picking something silly that we wouldn't actually end up wrapping in any of the JavaKit* modules, but yeah... Vector is extra silly. Thanks for the ArrayList suggestion.

Comment (ktoso) [reply Samples/JavaKitSampleApp/Sources/JavaKitExample/Java2Swift.config]: Yeah, no harm but a bit too silly maybe :) No prob

**Comment by DougGregor** [comment]
I have improvements I'd still like to make (e.g., handling Jar files, documenting this new path), but this is a good start

**Pull request merged**